test/test_datasets.py

```diff
@@ -344,3 +386,95 @@ def test_subset_samples_to_labels(labeled_datasets, labels):
         for sample in samples:
             assert dataset.sample_to_label[sample] in labels
         dataset.reset_filters()
+
+
+def test_set_all_data(all_datasets):
+    for dataset in all_datasets:
+        current_data = dataset.get_all_data()
+
+        if issubclass(type(dataset), datasets.LabeledDataset):
+            # Throw out labels in a labeled dataset
+            current_data = current_data[0]
+
+        new_data = np.random.normal(size=(1337, len(dataset.get_samples())))
+        print(new_data.shape)
+
+        dataset.set_all_data(new_data)
+
+        assert not np.array_equal(new_data, current_data)
+
+        retrieved_new_data = dataset.get_all_data()
+
+        if issubclass(type(dataset), datasets.LabeledDataset):
+            # Throw out labels in a labeled dataset
+            retrieved_new_data = retrieved_new_data[0]
+
+        assert np.array_equal(new_data, retrieved_new_data.T)
+
+
+def test_get_features(all_datasets):
+    for dataset in all_datasets:
+        features = dataset.get_features()
+
+        # Generate the gene names as in generate_test_data
+        gene_names = []
+        for i in range(generate_test_data.NUM_GENES):
+            gene_base = 'ENSG0000000000'
+            gene_name = '{}{}'.format(gene_base, i)
+            gene_names.append(gene_name)
+        assert features == gene_names
+
+
+def test_get_samples(all_datasets):
+    # Get sample names
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    expression_path = os.path.join(current_dir, 'data/test_expression.tsv')
+    with open(expression_path) as expression_file:
+        true_samples = set(expression_file.readline().strip().split('\t'))
+
+    for dataset in all_datasets:
+        print(type(dataset))
```

Comment:
Do you want to leave this print statement in?

Reply:
Good catch, went ahead and removed the debug print statements